src/Dotnet.Script.Core/ScriptCompiler.cs

```diff
@@ -190,8 +190,12 @@ private Assembly MapUnresolvedAssemblyToRuntimeLibrary(IDictionary<string, Runti
             {
                 if (runtimeAssembly.Name.Version > assemblyName.Version)
                 {
-                    Logger.Log($"Redirecting {assemblyName} to {runtimeAssembly.Name}");                    
-                    return Assembly.LoadFile(runtimeAssembly.Path);
+                    var loadedAssembly = AppDomain.CurrentDomain.GetAssemblies()
+                        .FirstOrDefault(a => a.GetName().Name == assemblyName.Name);
+                    if(loadedAssembly != null) 
```

Comment:
one final thing, for the sake of understanding what is going on, let's log here a message that we redirect to an already loaded assembly